### ISSUE TITLE
Finish CI and solve permissions issue with Docker

### ISSUE THREAD

**Issue by jlperla** [QuantEcon/lecture-source-jl#490]
After #489 we can try to put things together for CI.  Just remember that there is no persistence of any storage.

The sphinx generation of the lectures could happen with every CI run, as it is pretty fast.

In order to only execute the notebooks that have changed, we will need some sort of trick to find the modified notebooks and run them with #489 approaches.

Here are a few things to look at for how to analyze the git commit in the CI with `$TRAVIS_COMMIT_RANGE`
- https://reflectoring.io/skip-ci-build/
- https://github.com/exercism/dart/issues/63

**Comment by jlperla** [comment]
An alternative approach to try (first!) is with travis caches,

See https://docs.travis-ci.com/user/caching/  for more.

It may be possible to cache all of the sphinx output.  Then we only run the regerssion test on the notebooks that changed.

**Comment by jlperla** [comment]
One idea is that https://docs.travis-ci.com/user/caching/#arbitrary-directories shows we could have the whole` _build` directory cached?

The problem with this approach is that the amount of time to do the first build (and hence prepopulate the notebooks) is so long that I don't think it can finish in the time travis runs.  Not sure if that makes the approach untenable.

**Comment by arnavs** [comment]
@ajozefiak Judging from what you said, we're able to run a particular notebook in Docker and get its success/failure information? 

If so, then I think we can start testing with CI. The workflow (for a new PR or such) is: 

1. Run `make jupyter-tests`. This doesn't take too long, so it can happen for every file .
2. Execute (without `--allow-errors`) **only** those files that were modified in the PR.
3. Make sure that the exit status from that propagates down; i.e., Travis returns a pass if the notebooks run without errors, and a failure otherwise. 

Travis will start by checking out the PR under consideration, so we don't need to do any git operations (except maybe a `git diff` vs master). But maybe not: see the `TRAVIS_COMMIT_RANGE` variable that @jlperla mentioned in the issue.

Let me know if this makes sense, or if there's anything I can help clear up.

* While you're doing this, it makes sense to fork the repo and test with PRs against the fork, since the `master` here is used for production.
